### PR TITLE
Fix make on Mac replacing sed with tr

### LINKED ISSUE #28930: make on Mac breaks with a bad target for older sed
```
make generated_files DBG_MAKEFILE=1
Makefile:17: ***** starting makefile for goal(s) "generated_files"
Makefile:18: ***** Wed Jul 13 21:53:53 EDT 2016
Makefile.generated_files:93: ***** finding all *.go dirs
Makefile.generated_files:160: ***** finding all +k8s: tags
Makefile.generated_files:188: ***** finding all +k8s:deepcopy-gen tags
Makefile.generated_files:296: ***** finding all +k8s:conversion-gen tags
mkdir -p .make/_output/bin;                                        \
	(echo -n "_output/bin/conversion-gen: ";                        \
	 DIRECT=$(go list -e -f '{{.Dir}} {{.Dir}}/*.go' ./cmd/libs/go2idl/conversion-gen);                \
	 INDIRECT=$(go list -e   -f '{{range .Deps}}{{.}}{{"\n"}}{{end}}' ./cmd/libs/go2idl/conversion-gen | grep "^k8s.io/kubernetes" | xargs go list -e -f '{{.Dir}} {{.Dir}}/*.go');  \
	 echo $DIRECT $INDIRECT | sed 's/ / \\\n\t/g';       \
	) | sed "s|$(pwd -P)/||" > .make/_output/bin/conversion-gen.mk.tmp;                    \
	cmp -s .make/_output/bin/conversion-gen.mk.tmp .make/_output/bin/conversion-gen.mk || cat .make/_output/bin/conversion-gen.mk.tmp > .make/_output/bin/conversion-gen.mk && rm -f .make/_output/bin/conversion-gen.mk.tmp
mkdir -p .make/_output/bin;                                        \
	(echo -n "_output/bin/deepcopy-gen: ";                          \
	 DIRECT=$(go list -e -f '{{.Dir}} {{.Dir}}/*.go' ./cmd/libs/go2idl/deepcopy-gen);                  \
	 INDIRECT=$(go list -e   -f '{{range .Deps}}{{.}}{{"\n"}}{{end}}' ./cmd/libs/go2idl/deepcopy-gen | grep "^k8s.io/kubernetes" | xargs go list -e -f '{{.Dir}} {{.Dir}}/*.go');  \
	 echo $DIRECT $INDIRECT | sed 's/ / \\\n\t/g';       \
	) | sed "s|$(pwd -P)/||" > .make/_output/bin/deepcopy-gen.mk.tmp;                    \
	cmp -s .make/_output/bin/deepcopy-gen.mk.tmp .make/_output/bin/deepcopy-gen.mk || cat .make/_output/bin/deepcopy-gen.mk.tmp > .make/_output/bin/deepcopy-gen.mk && rm -f .make/_output/bin/deepcopy-gen.mk.tmp
make: *** No rule to make target `\nt/Users/clayton/projects/go/src/k8s.io/kubernetes/cmd/libs/go2idl/deepcopy-gen/main.go', needed by `_output/bin/deepcopy-gen'.  Stop.
```

I don't know where the `\nt` at the beginning of the target comes from, but probably from assumptions about sed version that are wrong.

I had to update to gnu-sed 4.2.2 to build.

### PR DESCRIPTION
**Pull request by aledbf**
fixes #28930

### CODE REVIEW

**Comment by aledbf** [comment]
@thockin fixes https://github.com/kubernetes/kubernetes/pull/28903#issuecomment-232517240

**Comment by dims** [comment]
Looks good to me 👍

**Comment by Gnouc** [comment]
@aledbf `tr '=,' \n\t'` save you one `tr` invocation.

**Comment by aledbf** [comment]
@Gnouc thanks for the suggestion

**Comment by MHBauer** [comment]
builds working binaries for me on osx, but I see a lot of stuff printed out.

File: Makefile.generated_files [hunk 1]
```diff
@@ -244,7 +244,7 @@ $(META_DIR)/$(DEEPCOPY_GEN).mk:
 	     ./cmd/libs/go2idl/deepcopy-gen                    \
 	     | grep "^$(PRJ_SRC_PATH)"                         \
 	     | xargs go list -e -f '{{.Dir}} {{.Dir}}/*.go');  \
-	 echo $$DIRECT $$INDIRECT | sed 's/ / \\\n\t/g';       \
+	  echo $$DIRECT $$INDIRECT | sed 's/ / =,/g' | tr '=,' '\n\t'; \
```
Comment (thockin) [inline Makefile.generated_files]: this is a bad indent.  it was right before :)

**Comment by thockin** [comment]
Thanks!

I made a comment, but I am LGTM'ing anyway.  You can send a followup or if you move quickly you can fix this one :)

File: Makefile.generated_files (hunk 1 above)
Comment (aledbf) [inline Makefile.generated_files]: done

**Comment by dims** [comment]
@k8s-oncall : why is this not getting merged?

**Pull request merged**